### PR TITLE
Preparing the CP Integration tests to use the Managed Seed implementation

### PR DESCRIPTION
**Pull request by krgostev**
**How to categorize this PR?**

/area control-plane-migration
/kind test

**What this PR does / why we need it**:
This PR removes the deprecated test definitions for registering and deleting seed. With the new Managed Seed implementation they are no longer needed. Also, this PR adjust the Shoot toleration in the migration test to the toleration that is used in the Managed Seed IT
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by timebertt** [comment]
/hold
until the `v1.19.0` release is cut.

**Review by rfranzke — COMMENTED** [review]
/invite @dguendisch @schrodit

File: test/framework/shootmigrationtest.go [hunk 1]
```diff
@@ -99,13 +99,14 @@ func (t *ShootMigrationTest) MigrateShoot(ctx context.Context) error {
 	t.MigrationTime = metav1.Now()
 	return t.GardenerFramework.MigrateShoot(ctx, &t.Shoot, t.TargetSeed, func(shoot *gardencorev1beta1.Shoot) error {
 		t := gardencorev1beta1.Toleration{}
-		t.Key = core.SeedTaintProtected
+		t.Key = SeedTaintTestRun
+		t.Value = pointer.StringPtr("12345") //GetTestRunID())
```
Comment (schrodit) [inline test/framework/shootmigrationtest.go]:
```suggestion
		t.Value = pointer.StringPtr("12345") //GetTestRunID())
```
what does the `GettestrunID` should the value be the id of the current testrun?
Can you add a comment to explain the value.

**Review by schrodit — CHANGES_REQUESTED** [review]

File: test/framework/shootmigrationtest.go (hunk 1 above)
Comment (krgostev) [reply test/framework/shootmigrationtest.go]: `GetTestRunID` returns the current testmachinery testrun ID.
The value should match the taint that is set in the seed management test [here](https://github.com/gardener/gardener/blob/bcdf52d75e2df05c6ff2ee49524964a8c2b05302/test/framework/managedseedframework.go#L262)
And yes, it should be `GetTestRunID()`. I have forgot to revert it after my local tests

**Review by schrodit — APPROVED** [review]

**Pull request merged**